File: java/my-intellij-plugin/src/dgroomes/CopyOpenTabsAction.java
package dgroomes;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.fileEditor.FileEditorManager;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.vfs.VirtualFile;
import com.intellij.openapi.ide.CopyPasteManager;
import java.awt.datatransfer.StringSelection;

public class CopyOpenTabsAction extends AnAction {

    @Override
    public void actionPerformed(AnActionEvent e) {
        Project project = e.getProject();
        if (project == null) return;

        FileEditorManager fileEditorManager = FileEditorManager.getInstance(project);
        VirtualFile[] openFiles = fileEditorManager.getOpenFiles();

        StringBuilder fileNames = new StringBuilder();
        for (VirtualFile file : openFiles) {
            fileNames.append(file.getName()).append("\n");
        }

        // Copy to clipboard
        CopyPasteManager.getInstance().setContents(new StringSelection(fileNames.toString()));
    }
}

File: java/my-intellij-plugin/src/dgroomes/CopyOpenTabsAction.kt
package dgroomes

import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class CopyOpenTabsAction : AnAction() {
    private val log: Logger = LoggerFactory.getLogger(CopyOpenTabsAction::class.java)

    override fun actionPerformed(e: AnActionEvent) {
        val project = e.project ?: run {
            log.error("Project was null.")
            return
        }

        val service = project.getService(CopyOpenTabsService::class.java)
        service.copyOpenTabNames()
    }
}
